File: slp/backend-nestjs/src/modules/admin/dto/admin-log-filter.dto.ts
export class AdminLogFilterDto {
  search?: string;          // free‑text search (username, action, targetType, targetId, details)
  action?: string;          // exact match
  targetType?: string;      // exact match
  from?: Date;
  to?: Date;
  adminId?: number;         // kept for future use
  count?: number;           // max number of logs to return, default 100
}

File: slp/backend-dotnet/Features/Admin/AdminLogFilterDto.cs
using System;

namespace backend_dotnet.Features.Admin;

public class AdminLogFilterDto
{
    public string? Search { get; set; }          // free‑text search
    public string? Action { get; set; }          // exact match
    public string? TargetType { get; set; }      // exact match
    public DateTime? From { get; set; }
    public DateTime? To { get; set; }
    public int? AdminId { get; set; }            // kept for future use, but not used by frontend
    public int? Count { get; set; } = 100;       // max number of logs to return
}